File: app/schemas/paper.py
import uuid
from typing import Any, List, Optional
from datetime import datetime
from pydantic import BaseModel, ConfigDict

class PaperResponse(BaseModel):
    id: uuid.UUID
    arxiv_id: str
    title: str
    abstract: str
    authors: List[dict[str, Any]]
    published_year: int
    pdf_url: str
    embedding: Optional[List[float]] = None
    arxiv_status: str
    created_at: datetime
    updated_at: datetime
    
    model_config = ConfigDict(from_attributes=True)

class PaperListResponse(BaseModel):
    items: List[PaperResponse]
    total: int

File: scientific-copilot-ui/lib/types/paper.ts
export type ArxivStatus = 'raw' | 'processed' | 'embedded' | 'extraction_failed'

export interface Paper {
  id: string
  arxiv_id: string
  title: string
  abstract: string
  authors: { name: string }[]
  published_year: number
  pdf_url: string
  embedding: number[] | null
  arxiv_status: ArxivStatus
  cluster_id: string | null
  created_at: string
  updated_at: string
}

export interface PaginatedPapers {
  items: Paper[]
  total: number
}
